Jake Storeapp step3

ChocOZerO opened the pull request.

ChocOZerO committed:
[Add]Sections #7

ChocOZerO committed:
[Add]Custom Cell for Header #8

ChocOZerO committed:
[Readme]정리 #9

ChocOZerO committed:
[Readme]이미지 사이즈 조절 #9

godrm commented on StoreApp/StoreApp/ViewControllers/ViewController.swift:
```diff
@@ -33,18 +38,37 @@ class ViewController: UIViewController, UITableViewDataSource {
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return CGFloat(60)
```
상수값도 의미있는 코드로 지정하세요.

godrm commented on StoreApp/StoreApp/ViewControllers/ViewController.swift:
```diff
@@ -33,18 +38,37 @@ class ViewController: UIViewController, UITableViewDataSource {
+            String($0.trimmingCharacters(in: [" "]))
+        }
+        headerView.titleLabel.text = header[0]
+        headerView.descriptionLabel.text = header[1]
```
header[0] 과 header[1] 에 대해서 의도를 알 수 있도록 표현해주세요.

godrm commented on StoreApp/StoreApp/ViewControllers/ViewController.swift:
```diff
@@ -33,18 +38,37 @@ class ViewController: UIViewController, UITableViewDataSource {
         guard let cell = tableView.dequeueReusableCell(withIdentifier: Keyword.customCellName.value, for: indexPath)
             as? StoreItemTableViewCell else { return UITableViewCell() }
-        guard let storeItem = storeItems[indexPath.row] else { return cell }
+        let storeItem = storeItems[indexPath.section][indexPath.row]
```
어떤 부분이 Delegate 메소드이고 어떤 부분이 DataSource 메소드인가요?
두 프로토콜이 어떤 용도로 어떻게 다른가 정리해보세요.

godrm replied:
이 부분은 피드백 반영한 내용이 뭐죠?

ChocOZerO replied:
주석이랑 Readme에 정리했습니다.

godrm replied:
그러면 extension 으로 나누는 구분은 어느 정도로 하면 될까요?

ChocOZerO replied:
추가 수정한 부분처럼 구분하면 어떨까요?

godrm replied:
항상 분리해야하는 건 아닙니다. 
구현해야 하는 메소드가 1-2개만 있으면 애매할때가 있죠. 그래서 물어봤습니다 :)

godrm replied:
다만 DataSource 객체를 분리하거나 Delegate 객체를 분리해야 하는 기준이 있다면 ViewController가 하지 않는게 더 좋을 수 있습니다.

ChocOZerO committed:
[Refactoring]피드백 적용 #10

- 상수값도 의미있는 코드로 지정하세요.
- header[0] 과 header[1] 에 대해서 의도를 알 수 있도록 표현해주세요.
- 어떤 부분이 Delegate 메소드이고 어떤 부분이 DataSource 메소드인가요? 두 프로토콜이 어떤 용도로 어떻게 다른가 정리해보세요.

ChocOZerO commented:
피드백 적용했습니다

ChocOZerO committed:
[Refactoring]DataSource와 Delegate 구분 #10
